specs/IFrameMemoryUsage

```diff
@@ -0,0 +1,251 @@
+      second_level_iframe_0 ...       second_level_iframe_n ...        [depth:2, second level iframe]
+               |
+              ...                                                      
+</pre> 
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` and `CoreWebView2Frame` to include the 
+`ProcessId` property and  `ProcessIdChanged` event. This property stores 
+the render process id for its iframe. This event will be raised whenever 
+the render process id is changed for `CoreWebView2` or `CoreWebView2Frame`.
+
+Additionally, we propose to add the `FrameCreated` event to `CoreWebView2Frame`. 
+This event fires when a new frame is created.
+
+
+# Examples
+C++
+```c++
+void ScenarioWebViewEventMonitor::InitializeEventView(ICoreWebView2* webviewEventView)
+{
+    m_webviewEventSource4->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [this](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    // ...
+                    // The new frame created event can be recursively initialized
+                    InitializeFrameEventView(webviewFrame, first_level_iframe_depth + 1);
+                    // ...
+                    return S_OK;
+                })
+                .Get(),
+            &m_frameCreatedToken);
+}
+
+void ScenarioWebViewEventMonitor::InitializeFrameEventView(
+    wil::com_ptr<ICoreWebView2Frame> webviewFrame, int depth) 
+{
+    auto frame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+    if (frame5)
+    {
+        //! [ICoreWebView2Frame:add_ProcessChanged]
+        frame5->add_FrameCreated(
+            Callback<ICoreWebView2NestedFrameCreatedEventHandler>(
+                [this, depth](
+                    ICoreWebView2Frame* sender,
+                    ICoreWebView2FrameCreatedEventArgs* args) -> HRESULT
+                {
+                    std::wstring message;
+                    wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webviewFrame));
+                    wil::unique_cotaskmem_string name;
+                    CHECK_FAILURE(webviewFrame->get_Name(&name));
+                    message += L"\"frame\": " + name.get());
+                    InitializeFrameEventView(webviewFrame, depth + 1);
+                    message += L",\"depth\": " + std::to_wstring(depth);
+                    auto webviewFrame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+                    if (webviewFrame5)
+                    {
+                        //! [ICoreWebView2Frame:get_ProcessId]
+                        unsigned int processId;
+                        CHECK_FAILURE(webviewFrame5->get_ProcessId(&processId));
+                        message +=
+                            L",\"render process id\": " + std::to_wstring((int)processId);

```

Comment:
We don't do anything with this message we create. Is there a real scenario we can demonstrate with the FrameCreated event? What did we do for the sample code in the CoreWebView2.FrameCreated spec?

Reply:
In `Win Sample App ->  Window Event Monitor`, we print out this message. I will add that to the spec.

Reply:
But what is the real world scenario? I guess we're adding it so apps can monitor performance and such via telemetry so this is perhaps fine.

Reply:
Apps can get memory usage based on process id.